src/main/java/org/lanternpowered/server/entity/LanternLiving.java

```diff
@@ -101,20 +131,90 @@ public void pulse() {
                     if (potionEffect.getType() == PotionEffectTypes.GLOWING) {
                         glowing = true;
                     }
+                    if (potionEffect.getType() == PotionEffectTypes.HUNGER && get(FoodData.class).isPresent()) {
+                        MutableBoundedValue<Double> exhaustion = get(FoodData.class).get().exhaustion();
+                        offer(Keys.EXHAUSTION, Math.min(exhaustion.get() + (0.005 * (potionEffect.getAmplifier() + 1)), exhaustion.getMaxValue()));
+                    }
                 }
             }
             offer(Keys.POTION_EFFECTS, newPotionEffects.build());
         }
         offer(Keys.GLOWING, glowing);
     }
 
-    @Override
-    public <T extends Projectile> Optional<T> launchProjectile(Class<T> projectileClass) {
-        return Optional.empty();
+    private void pulseFood() {
+        if (!get(FoodData.class).isPresent()) {
+            return;
+        }
+        final Difficulty difficulty = getWorld().getDifficulty();
+
```

Comment:
Try to reuse the `Value`s more instead of looking them up multiple times and messing them up with normal `get` functions.
For example: 
```offer(Keys.FOOD_LEVEL, Math.max(get(Keys.FOOD_LEVEL).get() - 1, getValue(Keys.FOOD_LEVEL).get().getMinValue()));```
to:
```
final MutableBoundedValue<Integer> foodLevel = getValue(Keys.FOOD_LEVEL).get();
offer(Keys.FOOD_LEVEL, Math.max(foodLevel.get() - 1, foodLevel.getMinValue()));
```